string topic
string msg_type